src/concern/mod.rs

```diff
@@ -39,34 +90,35 @@ pub enum ReadConcern {
     Custom(String),
 }
 
-impl PartialEq for ReadConcern {
-    fn eq(&self, other: &Self) -> bool {
-        self.as_str() == other.as_str()
+impl ReadConcernLevel {
+    pub(crate) fn from_str(s: &str) -> Self {
+        match s {
+            "local" => ReadConcernLevel::Local,
+            "majority" => ReadConcernLevel::Majority,
+            "linearizable" => ReadConcernLevel::Linearizable,
+            "available" => ReadConcernLevel::Available,
+            s => ReadConcernLevel::Custom(s.to_string()),
+        }
     }
-}
 
-impl ReadConcern {
-    /// Gets the string representation of the `ReadConcern`.
-    pub fn as_str(&self) -> &str {
-        match *self {
-            ReadConcern::Local => "local",
-            ReadConcern::Majority => "majority",
-            ReadConcern::Linearizable => "linearizable",
-            ReadConcern::Available => "available",
-            ReadConcern::Custom(ref s) => s,
+    /// Gets the string representation of the `ReadConcernLevel`.
+    pub(crate) fn as_str(&self) -> &str {
```

Comment:
this didn't seem necessary as part of the public API. If anything, maybe a `Display` implementation would be preferable. I could add it back if we think users would miss it though.